common/lib/xmodule/xmodule/capa_module.py

```diff
@@ -303,13 +303,174 @@ def max_score(self):
+    # @staticmethod
+    # def generate_report_data(descriptor):
+    # v3: non-static, and receive user_state_client as parameter
+    def generate_report_data(self, user_state_iterator, limit_responses=5000):
+        """
+        Return a list of student responses to this block in a readable way.
+        Each call returns a tuple like: ("username", {"Question": "2 + 2 equals how many?", "Answer": "Four"})
+        """
+
+        # import sys; sys.stdout = sys.__stdout__; import ipdb; ipdb.set_trace()
+
+        if limit_responses:
+            pass
+            # raise NotImplementedError("implement a limit")
+
+        if False and "testing with static":
+            # import sys; sys.stdout = sys.__stdout__; import ipdb; ipdb.set_trace()
+            self = descriptor
+
+            # from xmodule.tests import DATA_DIR, get_test_system, get_test_descriptor_system
+            # self.xmodule_runtime = get_test_system()
+            # assert self.xmodule_runtime
+
+        if self.category != 'problem':
+            raise NotImplementedError()
+        # FIXME reimplement without using self.lcp (to make it work with the static one)
+        if False and 'customresponse' in [elem.tag for elem in self.lcp.responder_answers.keys()]:
+            raise NotImplementedError("Not implemented for custom response problems "
+                                      "(like drag&drop, chemical equations etc.)")
+
+        # FIXME remove these tests:
+        # from opaque_keys.edx.keys import CourseKey, UsageKey
+        # block_key = UsageKey.from_string('block-v1:edX+DemoX+Demo_Course+type@problem+block@a0effb954cca4759994f1ac9e9434bf4')
+
+        block_key = self.location
+
+        # FIXME needed?
+        try:
+            tree = etree.XML(self.data)
+        except etree.XMLSyntaxError:
+            log.error('Error parsing problem types from xml for capa module {}'.format(self.display_name))
+            return
+
+
+        from capa.capa_problem import LoncapaProblem, LoncapaSystem
+        capa_system = LoncapaSystem(
+            ajax_url=None,
+            anonymous_student_id=None,
+            cache=None,
+            can_execute_unsafe_code=None,
+            get_python_lib_zip=None,
+            DEBUG=None,
+            filestore=self.runtime.resources_fs,
+            i18n=self.runtime.service(self, "i18n"),
+            node_path=None,
+            render_template=None,
+            seed=1,
+            STATIC_URL=None,
+            xqueue=None,
+            matlab_api_key=None,
+        )
+
+        # import sys; sys.stdout = sys.__stdout__; import ipdb; ipdb.set_trace()
+
+        # FIXME remove this step
+        states = list(user_state_iterator)
+        print("I am seeing the following user_state:", states)
+
+        for user_state in states:
```

Comment:
You can replace this with:
```python
for idx, user_state in enumerate(states):
    if idx >= limit_responses: break
```
I think that should be enough to support limit_responses.